services/adgs/rs_server_adgs/api/adgs_search.py

```diff
@@ -83,11 +84,25 @@ def __init__(self, request: Request | None = None, readwrite: Literal["r", "w"]
     @handle_exceptions
     async def process_search(self, collection: dict, odata_params: dict) -> stac_pydantic.ItemCollection:
         """Do the search for the given collection and OData parameters."""
+        # Priority, stac endpoint limit -> odata collection top -> 1000 by default
+        user_limit = int(self.request.query_params.get("limit", odata_params.get("top", 1000)))
+        if page := self.request.query_params.get("page"):
+            try:
+                # Attempt to parse `page` as an integer and validate it's >= 1
+                page = int(page)
+                if page < 1:
+                    raise ValueError("Page number invalid")
+
+            except (ValueError, TypeError) as exc:
+                # Handle cases where `page` is not an integer or is invalid
+                raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Page number invalid") from exc
         return process_product_search(
             collection.get("station", "adgs"),
             odata_params.get("productType"),
             odata_params.get("PublicationDate"),
-            odata_params.get("top", None),
+            user_limit,
+            self.request.query_params.get("sortby", "-created"),
```

Comment:
Is it necessary to set default value "-created", because in the process_product_search function below, this is already the default value. It would be better to use this default value at a single place to improve maintainability.

Reply:
Ok removed from signature